src/frontend/hooks/server/deviceInfo.ts

```diff
@@ -17,12 +22,12 @@ export const useEditDeviceInfo = () => {
   const queryClient = useQueryClient();
 
   return useMutation({
-    mutationKey: ['device'],
+    mutationKey: ['deviceInfo'],
     mutationFn: async (name: string) => {
       return mapeoApi.setDeviceInfo({name});
     },
-    onSuccess: () => {
-      queryClient.invalidateQueries({queryKey: ['deviceInfo']});
+    onSuccess: async () => {
+      return await queryClient.invalidateQueries({queryKey: ['deviceInfo']});
```

Comment:
This allows for optimistic updates. Essentially it forces the cache to to become valid only AFTER it has resolved the promise of the API call. Otherwise it is considered a valid cache as soon as the call is made, even before the promise has resolved.

Reply:
Do we want optimistic updates? Previously we would show a saving state (it was a disabled state of the save button for observation edit - maybe not the best UI), and then navigate on success. We had discussed showing a "save success" message, but never got round to implementing it.

It's arguable that otpimisitc updates are more trouble and complexity than they are worth for a local-only app, where all mutation actions are going to be very quick, and they could result in a confusing user experience: save -> appears to have updated -> then see error message.

I think we need more UX work on this overall, perhaps for all forms which are async having states for "saving", "error saving" and "saved". However in the meantime maybe it's not worth the work on optimistic updates given it doesn't give us a clear way to handle errors.

Reply:
https://github.com/digidem/CoMapeo-mobile/pull/137#issuecomment-1861325797